lib/src/config.dart

```diff
@@ -4,11 +4,51 @@
 
 library linter.src.config;
 
+import 'package:analyzer/plugin/options.dart';
 import 'package:yaml/yaml.dart';
 
+/// Process the given option [fileContents] and produce a corresponding
+/// [LintConfig].
+LintConfig processAnalysisOptionsFile(String fileContents, {String fileUrl}) {
+  var yaml = loadYamlNode(fileContents, sourceUrl: fileUrl);
+  if (yaml is YamlMap) {
+    return _parseConfig(yaml);
+  }
+  return null;
+}
+
+LintConfig _parseConfig(Map optionsMap) {
+  if (optionsMap != null) {
+    var options = optionsMap['linter'];
```

Comment:
Assuming options processors are associated with plugins and that plugins will be published via pub I was thinking that the pub namespace would solve this for us.  Does that sound right?